plugins/playlist/src/components/PlaylistEditDialog/PlaylistEditDialog.tsx

```diff
@@ -74,13 +75,39 @@ export const PlaylistEditDialog = ({
 }: PlaylistEditDialogProps) => {
   const classes = useStyles();
   const identityApi = useApi(identityApiRef);
+  const playlistApi = useApi(playlistApiRef);
+  const [playlistArray, setPlaylistArray] = useState<PlaylistMetadata[]>([]);
+  const [editingOtherFields, setEditingOtherFields] = useState(false);
+
+  useEffect(() => {
+    const fetchPlaylists = async () => {
+      const playlists = await playlistApi.getAllPlaylists({ editable: true });
+      setPlaylistArray(playlists);
+    };
+
+    fetchPlaylists();
+  }, [playlistApi]);
 
   const { loading: loadingOwnership, value: ownershipRefs } =
     useAsync(async () => {
       const { ownershipEntityRefs } = await identityApi.getBackstageIdentity();
       return ownershipEntityRefs;
     }, []);
 
+  const nameIsUnique = async (
+    name: string,
+    isEditing: boolean,
+    originalName: string,
+  ) => {
+    if (!isEditing || name !== originalName) {
+      return playlistArray.some(p => p.name === name)
```

Comment:
Now there's a race condition instead, such that if the playlist hasn't loaded yet it can't validate properly (it'll just see the empty array). What I suggested was to call `getAllPlaylists` INSIDE here, without a `useState`.

But alright you know what, if we want to keep the just-do-one-request pattern, you can just do `const playlistPromise = useRef(playlistApi.getAllPlaylists({ editable: true }));` and delete the `useEffect`. Then it's is a `Promise<PlaylistMetadata[]>`, so your `nameIsUnique` function can do `const playlist = await playlistPromise.current;` to get the actual value. That resolves the race